### ISSUE TITLE
Add support for AzureOpenAI

### ISSUE THREAD

**Issue by rajib76** [whylabs/langkit#197]
I see that langkit is not supporting Azure Open AI. When can this be supported?

**Comment by jamie256** [comment]
Hi @rajib76! Thanks for opening an issue. 

Are you looking for support of an Azure OpenAI deployed LLM in LangKit's [OpenAIDefault ](https://github.com/whylabs/langkit/blob/main/langkit/openai/openai.py#L130) kind of thing like we do in the [Choosing an LLM](https://github.com/whylabs/langkit/blob/main/langkit/examples/Choosing_an_LLM.ipynb) example? or something else? 

I wanted to get some better support for Azure hosted models into LangKit soon, we could probably focus on changes to support the Azure OpenAI models as a first iteration if that is helpful (e.g. gpt-35-turbo, gpt4)?

**Comment by rajib76** [comment]
Yes, looking for support in LangKit's OpenAIDefault. Currently if I need to do hallucination checks, looks like I cannot do it using Azure Open AI. It is only supported for open ai. I am looking at using to evaluate the response from Azure Open AI for hallucination, prompt injection, contextual relevancy and all

**Comment by jamie256** [comment]
Ok, working on it. If you want to try an initial dev build:
`pip install langkit==0.0.26.dev0`

New class and usage looks like this:
```
from langkit import response_hallucination
from langkit.openai import OpenAIAzure

response_hallucination.init(llm=OpenAIAzure(engine="LangKit-test-01"), num_samples=1)
```

Also need to set these new env vars:
```
os.environ["AZURE_OPENAI_ENDPOINT"]
os.environ["AZURE_OPENAI_KEY"]
```

As referenced in this example: https://learn.microsoft.com/en-us/azure/ai-services/openai/how-to/chatgpt?tabs=python&pivots=programming-language-chat-completions#working-with-the-gpt-35-turbo-and-gpt-4-models

**Comment by rajib76** [comment]
Was able to run it but saw one thing. I tried the example code for hallucination score. I did not see a way to add a context and then tell the LLM to grade the response based on question and context

**Comment by rajib76** [comment]
> Was able to run it but saw one thing. I tried the example code for hallucination score. I did not see a way to add a context and then tell the LLM to grade the response based on question and context

I think I got how it is working. It is self check validation. The prompt is being sent to the same model to get and answer again and then we are checking back with the response. Is it possible to do below
1. Send a ground truth instead of LLM creating sample
2. If I need to create a sample with a llm, can I use a different LLM than the one which actually does the hallucination check(or is the hallucination check done by the LLM or through a ML model)

**Comment by FelipeAdachi** [comment]
Thanks for the reply @rajib76 

For #1, yes, it should be possible to perform the semantic similarity based consistency check without the presence of an LLM.

And #2 also makes a lot of sense for your and others' scenarios

I created two issues to reflect both topics we are discussing:

- https://github.com/whylabs/langkit/issues/205

- https://github.com/whylabs/langkit/issues/204

We'll plan those changes in future sprints

**Comment by ishachinniah-hds** [comment]
Hi @jamie256, 

Is Azure Open AI support now integrated with the latest version of langkit? I am trying to use the response_halucination module in the following way but am getting these error messages:

### Langkit response_halucination initialization 
```
    # LANGKIT
    response_hallucination.init(llm=OpenAIAzure(), num_samples=1)
    result = response_hallucination.consistency_check(
        prompt=query,
        response=response,
    )
    return result
```

### Setting environment variables from Azure Open AI generated api keys
```
os.environ["AZURE_OPENAI_ENDPOINT"] = config.AZURE_OPENAI_ENDPOINT
os.environ["AZURE_OPENAI_KEY"] = config.OPENAI_API_KEY
```

### Error traceback
result = response_hallucination.consistency_check(
  File "/opt/homebrew/anaconda3/envs/chatbot/lib/python3.9/site-packages/langkit/response_hallucination.py", line 278, in consistency_check
    return checker.consistency_check(prompt, response).to_summary_dict()
  File "/opt/homebrew/anaconda3/envs/chatbot/lib/python3.9/site-packages/langkit/response_hallucination.py", line 223, in consistency_check
    additional_samples: List[ChatLog] = self.get_samples(prompt)
  File "/opt/homebrew/anaconda3/envs/chatbot/lib/python3.9/site-packages/langkit/response_hallucination.py", line 109, in get_samples
    raise Exception(
Exception: Response Hallucination - Error generating sample: Error code: 404 - {'error': {'code': '404', 'message': 'Resource not found'}}

### Next Steps
Could you let me know what the current method for integrating azure openai llms for evaluations in langkit is? Let me know if there is additional information I can provide. Thank you
